sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -560,44 +597,60 @@ object GpuCast extends Arm {
    * Asserts that all values in a column are within the specific range.
    *
    * @param values ColumnVector to be performed with range check
-   * @param minValue Named parameter for function to create Scalar representing range minimum value
-   * @param maxValue Named parameter for function to create Scalar representing range maximum value
+   * @param minValue Range minimum value of input type T
+   * @param maxValue Range maximum value of input type T
    * @param inclusiveMin Whether the min value is included in the valid range or not
    * @param inclusiveMax Whether the max value is included in the valid range or not
    * @param errorMsg Specify the message in the `IllegalStateException`
    * @throws IllegalStateException if any values in the column are not within the specified range
    */
-  private def assertValuesInRange(values: ColumnView,
-      minValue: => Scalar,
-      maxValue: => Scalar,
+  private def assertValuesInRange[T](values: ColumnView,
+      minValue: T,
+      maxValue: T,
```

Comment:
We can improve a little bit:
```suggestion
  private def assertValuesInRange[T, U](values: ColumnView,
      minValue: U,
      maxValue: U,
```

Then at the line 652 below:
```
        throwIfOutOfRange(getValue(minInput), minValue.asInstanceOf[T], getValue(maxInput), maxValue.asInstanceOf[T])
```

Reply:
Thank you! One thing I found is that `Int.MaxValue.asInstanceOf[BigDecimal]` throws an error. If `maxValue` is `Int.MaxValue` and `T` is `BigDecimal` then we would run into this issue.

Reply:
Just pass in `BigDecimal(Int.MaxValue)`.

Reply:
Then why do we want to pass in type T and U if they are the same type? Sorry I might have misunderstood.

Reply:
They are the same type if `T` is `BigDecimal`.

Reply:
I just ran the `AnsiCastOpTestSuite`. I had 20 failed tests. An example error is```
 java.lang.Short cannot be cast to java.lang.Long
	at scala.runtime.BoxesRunTime.unboxToLong(BoxesRunTime.java:107) ~[scala-library-2.12.15.jar:?]
	at scala.math.Ordering$Long$.compare(Ordering.scala:294) ~[scala-library-2.12.15.jar:?]
	at com.nvidia.spark.rapids.GpuCast$.throwIfOutOfRange$1(GpuCast.scala:627) ~[classes/:?]
```.  It seems like the implicit ordering `compare` function is complaining.

Reply:
It seems `Short` can't be implicitly cast to `Long`. In such cases, you need to cast the parameter: `Short.MinValue.toLong`.

Reply:
I also have `Byte cannot be cast to Short/Int`, `Byte/Short cannot be cast to Int`, `Byte/Short/Int cannot be cast to Long`, `Long/Int/Short/Byte cannot be cast to float/double`. If I cast these explicitly, I am afraid I would go back to the original solution.